app/src/main/java/com/example/euriskocodechallenge/ui/login/SignUpActivity.kt

```diff
@@ -30,37 +31,47 @@ class SignUpActivity : AppCompatActivity() {
         setContentView(binding.root)
         supportActionBar?.hide()
 
-        //Check if fields are valid, then insert new user
-        //We can also add Check if User exist with same email address
+
+        implementListeners()
+        initObservers()
+    }
+
+    private fun implementListeners() {
         binding.btnSignup.setOnClickListener {
             if (validateFields() && validatePass()) {
-                val user = User(
-                    userId = 0L,
-                    email = binding.emailEt.text?.trim().toString(),
-                    fName = binding.fNameEt.text?.trim().toString(),
-                    lName = binding.lNameEt.text?.trim().toString(),
-                    password = binding.passEt.text?.trim().toString(),
-                    application.getDrawable(R.drawable.ic_account_circle)!!.toBitmap()
-                )
-                userViewModel.insertUser(user)
+                val user =
+                    application.getDrawable(R.drawable.ic_account_circle)?.toBitmap()?.let { it1 ->
+                        User(
+                            userId = 0L,
+                            email = binding.emailEt.text?.trim().toString(),
+                            fName = binding.fNameEt.text?.trim().toString(),
+                            lName = binding.lNameEt.text?.trim().toString(),
+                            password = binding.passEt.text?.trim().toString(),
+                            it1
+                        )
+                    }
+                user?.let {
+                    userViewModel.insertUser(user)
+                }
             }
         }
-        observeViewModel()
     }
 
-    //This function handles all the LiveData to be observed
-    private fun observeViewModel() {
+
+    //Check if fields are valid, then insert new user
+    //We can also add Check if User exist with same email address
+    private fun initObservers() {
         //On Successful Login, Redirect to HomeActivity and setUserLoggedIn in DataStore
         userViewModel.fetchUser().observe(this) {
-            Toast.makeText(this, "Welcome ${it.fName} ${it.lName}", Toast.LENGTH_LONG).show()
+            utilityfunctions.showtoast(this, "Welcome ${it.fName} ${it.lName}")
             startActivity(Intent(this, HomeActivity::class.java))
         }
         //- Can Be Removed - Checks if user inserted successfully
         userViewModel.fetchInsertedId().observe(this) {
             if (it != -1L && it != 0L) {
-                Toast.makeText(this, "Registration Successful", Toast.LENGTH_LONG).show()
+                utilityfunctions.showtoast(this, "Registration Successful")
```

Comment:
@martin371998 hardcoded strings should be in strings.xml file

Reply:
fixed in Utility functions new PR